### DISCUSSION TITLE
After nodes prediction I can't get those results in hdf5 file

### DISCUSSION THREAD

**Discussion by GioLetti** [talmolab/sleap#1782] [Help!] [answered]
Hi,

I started using sleap to perform some tracking analysis and I found the software great, in particular the GUI is really great. Great job. 

I have just a small problem, I can't get the prediction data in the hdf5 file.

In particular, I'm using the single_animal model to track a single animal in the video. I perform the training and the inference, selecting even the flow tracker in the inference window of the GUI. But after inference I get just the instance prediction with no track prediction, and when I save the HDF5 analysis file the 'track' dataset is empty.

I can get the track by using the command 'set instance track' of Tracks widget in the menu bar, but I should do this for each frame of the movie. 

I already solved the problem by using the command sleap-track directly from the terminal, but is there a way to do so with a single command from the GUI?

Moreover, the problem does not appear if instead of the single animal model I use the top-down model.

Am I doing something wrong?

Help is much appreciated.

Giorgio

**Comment by roomrys** [comment, marked as answer]
Hi @GioLetti,

For single animal projects, we suggest not using `Track`s at all.  The main purpose of `Track`s is to sort which pose estimation/annotation belongs to which animal throughout an entire video. However, if there is just a single pose estimation/annotation in each `LabeledFrame`, then we already know that the pose estimation belongs to the one and only animal being annotated (i.e. single animal project).

However, the logic for checking whether the user has a single animal project is not as direct as looping through all the user's labels to check that there are not more than 1 annotation per frame. Instead, we (in this case, incorrectly) assume that the `Labels`/slp project will have no `Track`s to save time - or, in cases like yours, cause headaches.

In the GUI, can you Track > Delete Multiple Tracks > Unused. Save As. Retry the export.

Sorry for the troubles! Please let us know if that helped.

Thanks,
Liezl
